builder/googlecompute/driver_gce.go

```diff
@@ -168,13 +168,21 @@ func NewDriverGCE(config GCEDriverConfig) (Driver, error) {
 	}, nil
 }
 
-func (d *driverGCE) CreateImage(name, description, family, zone, disk string, image_labels map[string]string, image_licenses []string, image_encryption_key *compute.CustomerEncryptionKey, imageStorageLocations []string) (<-chan *Image, <-chan error) {
+func (d *driverGCE) CreateImage(name, description, family, zone, disk string, image_labels map[string]string, image_licenses []string, image_guest_os_features []string, image_encryption_key *compute.CustomerEncryptionKey, imageStorageLocations []string) (<-chan *Image, <-chan error) {
+
+	imageFeatures := make([]*compute.GuestOsFeature, len(image_guest_os_features))
```

Comment:
I think that you meant:
```suggestion
	imageFeatures := make([]*compute.GuestOsFeature, len(image_guest_os_features), 0)
```

here. Example: https://go.dev/play/p/wCsKwsQiQMg

Reply:
Hello,

Thank you very much for your review!
Yes, I just copied the existing code for metadata and inherited the same error:

https://github.com/hashicorp/packer-plugin-googlecompute/blob/main/builder/googlecompute/driver_gce.go#L431
```
	metadata := make([]*compute.MetadataItems, len(c.Metadata))
```

But I think we need to use 0 for length and len() for capacity:
```
make([]*compute.GuestOsFeature, 0, len(image_guest_os_features))
```

Please take a look my example: https://go.dev/play/p/WK3FgRxBhSa

So please let me me fix my pull request!

And what about the metadata code ? Do we need fix it too ?

Thank you!

Reply:
Hello,

Please review 3fd0bb0:

    * Fixed GuestOsFeatures list
    * Added test to verify that the slice is correctly created.

Thank you!

Reply:
> But I think we need to use 0 for length and len() for capacity:

Oh, Yes, you are right, I typed that one a little bit too fast :D, hopefully, the compiler screams in that case 🙂 

Yeah, fixing the metadata would be a good thing I think too 👍🏼

Reply:
Hello,

Please tell me - can I fixed the metadata issue using this pull request ?
Or should I open a new pull request ?

Thank you!

Reply:
Ah - my pull request has been successfully merged. So please let me open a separate request for the metadata issue!

Thank you!